packages/cms-cli/commands/auth.js

```diff
@@ -41,14 +43,30 @@ async function authAction(type, options) {
 
   trackCommandUsage(COMMAND_NAME);
   let configData;
+  let updatedConfig;
+  let promptAnswer;
   switch (authType) {
     case OAUTH_AUTH_METHOD.value:
       configData = await promptUser(OAUTH_FLOW);
       await authenticateWithOauth(configData);
       break;
     case PERSONAL_ACCESS_KEY_AUTH_METHOD.value:
       configData = await personalAccessKeyPrompt();
-      await updateConfigWithPersonalAccessKey(configData);
+      updatedConfig = await updateConfigWithPersonalAccessKey(configData);
+
+      if (!updatedConfig.name) {
+        promptAnswer = await promptUser([PORTAL_NAME]);
+        updatePortalConfig({
+          ...updatedConfig,
+          environment: updatedConfig.env,
+          tokenInfo: updatedConfig.auth.tokenInfo,
+          name: promptAnswer.name,
```

Comment:
I think we should add some validation somewhere of the name to make sure it is not a duplicate and that it doesn't have spaces.

Reply:
@miketalley I recently added a validate function where you can add this.
https://github.com/HubSpot/hubspot-cms-tools/blob/d98f0449e0add7dc37205d1c8b614f6cb66256d1/packages/cms-lib/lib/config.js#L31
It works with the config data in memory whereas `updatePortalConfig` both updates and writes.  So something would need refactored for the user flow.  Like passing an optional `write: false` option to the update method.

Reply:
do we want to require a name for all portals? right now it is optional, and the `hs auth oauth2` flow does not include asking for a portal name.

personally, I like using it and do 99% of the time, but some developers may not like having it be required. it would cause errors for people with configs that have nameless portals. thoughts @corymartin @gcorne ?

Reply:
I added name validation for portals into the validateConfig method to handle duplicate portal names and names with spaces

Reply:
In a similar fashion to a form input, I think we should be validating during the prompt flow as it provides a better user experience.

Reply:
We have [validation with the prompt to disallow names that contain spaces](https://github.com/HubSpot/hubspot-cms-tools/blob/46c5945fe82259b354c3aee864637628b9ee5a23/packages/cms-cli/lib/prompts.js#L62). Do we want to add something else?

Reply:
I was thinking that we could validate on duplicate names in the prompt flow.

Reply:
I added a check to ensure a duplicate name will not be accepted when running `hs auth`